crates/sui/src/sui_commands.rs

```diff
@@ -112,6 +113,7 @@ pub enum SuiCommand {
 
 impl SuiCommand {
     pub async fn execute(self) -> Result<(), anyhow::Error> {
+        move_package::package_hooks::register_package_hooks(Box::new(SuiPackageHooks {}));
```

Comment:
Re https://github.com/MystenLabs/sui/pull/8778#discussion_r1122995780: 

> I think this is a fine place to put it when it comes to our CLI commands, but we will also have to think about for tests (we have tests that need to publish packages). Is there a way to cater to those as well? There may not be, in which case we'll just need to be careful about adding the hooks in all the right places.

I'm not sure yet how tests will be impacted, so right now I'm going with registering the custom hook in the basic unit test for this PR. When we have functionality around `published-at`, I'll investigate how it affects existing tests and then decide either (a) add hooks in the right places or (b) see if we can generalize this to wrapper harness, which is [what I see done elsewhere](https://sourcegraph.com/github.com/aptos-labs/aptos-core@ee8e02b/-/blob/aptos-move/e2e-move-tests/src/harness.rs?L68-71&subtree=true).